src/vstest.console/Processors/RunTestsArgumentProcessor.cs

```diff
@@ -236,7 +238,7 @@ private void TestRunRequest_OnRunCompletion(object sender, TestRunCompleteEventA
                     var testsFoundInAnySource = (e.TestRunStatistics == null) ? false : (e.TestRunStatistics.ExecutedTests > 0);
 
                     // Indicate the user to use testadapterpath command if there are no tests found
-                    if (!testsFoundInAnySource && string.IsNullOrEmpty(CommandLineOptions.Instance.TestAdapterPath))
+                    if (!testsFoundInAnySource && string.IsNullOrEmpty(CommandLineOptions.Instance.TestAdapterPath) && this.commandLineOptions.TestCaseFilterValue == null) 
```

Comment:
string.IsNullOrEmpty instead of == null?

Reply:
`TestCaseFilterValue` can't be empty because we won't accept empty value.